### PR TITLE
[shopsys] moved category parameters to framework package 

### PR DESCRIPTION
**Pull request by stanoMilan**
Move Category parameters from project base to framework

### CODE REVIEW

**Review by TomasLudvik — APPROVED** [review]

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 1]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
 
         return $category === null ? '' : $category->getName($domainLocale);
     }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param \App\Model\Category\Category|null $category
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php]:
```suggestion
     * @param \Shopsys\FrameworkBundle\Model\Category\Category|null $category
```
App namespace must not be used in the framework

Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php]:
```suggestion
     * @param \Shopsys\FrameworkBundle\Model\Category\Category|null $category
```
App namespace must not be used in the framework

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 2]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
...
+     */
+    protected function buildFilterParameters(FormBuilderInterface $builder, ?\App\Model\Category\Category $category): void
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php]: use `Shopsys\FrameworkBundle\Model\Category\Category`, please

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 3]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
...
+    {
+        if ($category === null) {
+            return;
+        }
+        $parametersFilterBuilder = $builder->add('parametersGroup', GroupType::class, ['label' => t('Filter parameters')]);
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php]: the translations should be moved in the po files as well

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 4]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
...
+
+        $parametersUsedByProductsInCategory = $this->parameterRepository->getParametersUsedByProductsInCategory($category, $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID));
+
+        foreach ($parametersUsedByProductsInCategory as $parameter) {
+            $parameterNamesById[$parameter->getId()] = $parameter->getName();
+        }
+
+        $parametersFilterBuilder->add('parametersPosition', SortableValuesType::class, [
+            'labels_by_value' => $parameterNamesById,
+            'label' => t('Parameters order in category'),
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php:290]: the translations should be moved in the po files as well

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 5]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
...
+            'required' => false,
+        ]);
+
+        $parametersFilterBuilder->add('parametersCollapsed', ChoiceType::class, [
+            'required' => false,
+            'label' => t('Filter parameters closed by default'),
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php:296]: the translations should be moved in the po files as well

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 6]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
...
+            $parameterNamesById[$parameter->getId()] = $parameter->getName();
+        }
+
+        $parametersFilterBuilder->add('parametersPosition', SortableValuesType::class, [
+            'labels_by_value' => $parameterNamesById,
+            'label' => t('Parameters order in category'),
+            'required' => false,
+        ]);
+
+        $parametersFilterBuilder->add('parametersCollapsed', ChoiceType::class, [
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php:294]: `CategoryData::$parametersCollapsed` property is still defined in the project-base, it should be moved as well, right?

File: packages/framework/src/Form/Admin/Category/CategoryFormType.php [hunk 7]
```diff
@@ -257,4 +261,40 @@ private function getCategoryNameForPlaceholder(DomainConfig $domainConfig, ?Cate
...
+
+        $parameterNamesById = [];
+
+        $parametersUsedByProductsInCategory = $this->parameterRepository->getParametersUsedByProductsInCategory($category, $this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID));
+
+        foreach ($parametersUsedByProductsInCategory as $parameter) {
+            $parameterNamesById[$parameter->getId()] = $parameter->getName();
+        }
+
+        $parametersFilterBuilder->add('parametersPosition', SortableValuesType::class, [
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Category/CategoryFormType.php:288]: `CategoryData::$parametersPosition` property is still defined in the project-base, it should be moved as well, right?

File: packages/framework/src/Model/Product/Parameter/ParameterRepository.php [hunk 8]
```diff
@@ -79,6 +83,42 @@ public function getById($parameterId)
         return $parameter;
     }
 
+    /**
+     * @param \App\Model\Category\Category $category
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Product/Parameter/ParameterRepository.php]: framework namespace, please

File: packages/framework/src/Model/Product/Parameter/ParameterRepository.php [hunk 9]
```diff
@@ -79,6 +83,42 @@ public function getById($parameterId)
...
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig $domainConfig
+     * @return \App\Model\Product\Parameter\Parameter[]
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Product/Parameter/ParameterRepository.php]: framework namespace, please

File: packages/framework/src/Model/Product/Parameter/ParameterRepository.php [hunk 10]
```diff
@@ -79,6 +83,42 @@ public function getById($parameterId)
...
+            ->groupBy('p, pt');
+
+        $this->applyCategorySeoConditions($queryBuilder, $category, $domainConfig->getId());
+
+        return $queryBuilder->getQuery()->execute();
+    }
+
+    /**
+     * @param \Doctrine\ORM\QueryBuilder $queryBuilder
+     * @param \App\Model\Category\Category $category
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Product/Parameter/ParameterRepository.php]: framework namespace, please

File: packages/framework/src/Model/Product/Parameter/ParameterRepository.php [hunk 11]
```diff
@@ -79,6 +83,42 @@ public function getById($parameterId)
...
+     * @param int $domainId
+     */
+    protected function applyCategorySeoConditions(QueryBuilder $queryBuilder, Category $category, int $domainId): void
+    {
+        $queryBuilder
+            ->join(\App\Model\Product\Product::class, 'product', Join::WITH, 'ppv.product = product')
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Product/Parameter/ParameterRepository.php]: framework namespace, please

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi, please check the failing standards + my comments

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**